templates/admin/user/list.tmpl

```diff
@@ -10,7 +10,67 @@
+							<label class="item"><input type="radio" name="status_filter[is_active]" value="0"> {{.i18n.Tr "admin.users.list_status_filter.not_active"}}</label>
+							<div class="ui divider"></div>
+							<label class="item"><input type="radio" name="status_filter[is_restricted]" value="0"> {{.i18n.Tr "admin.users.list_status_filter.not_restricted"}}</label>
+							<label class="item"><input type="radio" name="status_filter[is_restricted]" value="1"> {{.i18n.Tr "admin.users.list_status_filter.is_restricted"}}</label>
+							<div class="ui divider"></div>
+							<label class="item"><input type="radio" name="status_filter[is_prohibit_login]" value="0"> {{.i18n.Tr "admin.users.list_status_filter.not_prohibit_login"}}</label>
+							<label class="item"><input type="radio" name="status_filter[is_prohibit_login]" value="1"> {{.i18n.Tr "admin.users.list_status_filter.is_prohibit_login"}}</label>
+							<div class="ui divider"></div>
+							<label class="item"><input type="radio" name="status_filter[is_2fa_enabled]" value="1"> {{.i18n.Tr "admin.users.list_status_filter.is_2fa_enabled"}}</label>
+							<label class="item"><input type="radio" name="status_filter[is_2fa_enabled]" value="0"> {{.i18n.Tr "admin.users.list_status_filter.not_2fa_enabled"}}</label>
+						</div>
+					</div>
+
+					<!-- Sort Menu Item -->
+					<div class="ui dropdown type jump item">
+						<span class="text">
+							{{.i18n.Tr "repo.issues.filter_sort"}} {{svg "octicon-triangle-down" 14 "dropdown icon"}}
+						</span>
+						<div class="menu">
+							<button class="item" name="sort" value="oldest">{{.i18n.Tr "repo.issues.filter_sort.oldest"}}</button>
+							<button class="item" name="sort" value="newest">{{.i18n.Tr "repo.issues.filter_sort.latest"}}</button>
+							<button class="item" name="sort" value="alphabetically">{{.i18n.Tr "repo.issues.label.filter_sort.alphabetically"}}</button>
+							<button class="item" name="sort" value="reversealphabetically">{{.i18n.Tr "repo.issues.label.filter_sort.reverse_alphabetically"}}</button>
+							<button class="item" name="sort" value="recentupdate">{{.i18n.Tr "repo.issues.filter_sort.recentupdate"}}</button>
+							<button class="item" name="sort" value="leastupdate">{{.i18n.Tr "repo.issues.filter_sort.leastupdate"}}</button>
+						</div>
+					</div>
+				</div>
+
+				<!-- Search Text -->
+				<div class="ui fluid action input" style="max-width: 70%;">
+					<input name="q" value="{{.Keyword}}" placeholder="{{.i18n.Tr "explore.search"}}..." autofocus>
+					<button class="ui blue button">{{.i18n.Tr "explore.search"}}</button>
+				</div>
+
+				{{/* here we have valid go template syntax, but eslint doesn't like it and reports "error  Parsing error: Unexpected token {" */}}
+				<script>
+					<!-- /* eslint-disable */ -->
+					(function() {
+						window.giteaContext = window.giteaContext || {};
+						window.giteaContext.adminUserListSearchForm = {
```

Comment:
please use `window.config`, which is populated in https://github.com/go-gitea/gitea/blob/94f529af02e74f5d13fefda48d3ef9db4559f71f/templates/base/head.tmpl?plain=1#L29

Reply:
@noerw  Thank you for your review. I have saw `window.config` but I didn't choose it, because `window.config` is obviously for config (read-only) values. But I want to pass dynamic data which are not `config` related.

And I found other pages use `window.ActivityTopAuthors` to pass data into Vue engine: https://github.com/go-gitea/gitea/blob/ab77a24f18ae45248bd6c71778fe3bfacacf864f/templates/repo/activity.tmpl#L113 

That's why I said the frontend framework seems a little messy.

In my opinion, we should keep `window.config` for config-related data (or refactor it to `window.giteaConfig`, and use a dedicated object like `window.giteaContext` to pass data to javascript code. If you feel OK, I can help to refactor the legacy  `window.ActivityTopAuthors` and do more.

Reply:
I made a new commit, now the data is passed in `window.config.PageData`